### ISSUE TITLE
Add channels

### ISSUE THREAD

**Issue by rasviitanen** [rasviitanen/rustysignal#10]
Channels should be added to the server. A user could register for a channel number (i.e. 5678), and all nodes connected to that channel would receive messages from every member of the channel. Using weak pointers, the channel could automatically die when everyone is disconnected.

**Comment by kawpii** [comment]
@rasviitanen Side question: is this repo maintained? can this package be used in production to facilitate audio, video, text conversations?

**Comment by rasviitanen** [comment]
@kaushalyap absolutely, I use it for syncing browser storage between my devices and fix errors in this project as they occur (it has been a while since the last one). I should note that the master-branch has push-funcionality, meaning that you can send a notification to a user without requiring them to have your page open. I am currenly working on this, so it is only available for non-ssl and still very hacky and untested. If you dont need the push-functionality, I would suggest to use the version available from cargo (see readme on how to install it with SSL). Oh and also, the usernames are removed once you disconnect, so you need to use temporary usernames (e.g. UUID) for the connections.

If you need any help, I am happy to help.

If channels are important to you, I could probably fix it within a day.

I should also note that I have not yet tested it for heavly loads, so I have no idea how many connections it could handle.
